### ISSUE TITLE
Multiple invalid summaries in torch.nn documentation page

### ISSUE THREAD

**Issue by lithuak** [pytorch/pytorch#70498]
### 🐛 Describe the bug

Many summaries (descriptions) for [Non-linear Activation](https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity) section of torch.nn [documentation page](https://pytorch.org/docs/stable/nn.html) are invalid.

For instance, all of the summary for nn.ELU is "Applies the element-wise function:" missing the exact function it refers to, although the full version of the description on [nn.ELU's page](https://pytorch.org/docs/stable/generated/torch.nn.ELU.html#torch.nn.ELU) contains the correct description.

The problem is due to both the way Sphinx' `autosummary` extension extracts the summary from a docstring (only the first line/paragraph gets through) and to the style of docstrings for Non-linear Activation entries which doesn't take this behavior into account.

It's not clear which way would be the best to take here: either to tinker with Sphinx to make it render more than one paragraph of a docstring (not an exactly ease task!) or to rewrite the docstrings with respect to this. Or may be both. 

For instance, the `nn.RReLU` summary that says: "Applies the randomized leaky rectified liner unit function, element-wise, as described in the paper:" can be fixed by just correcting the URL of the mentioned paper to be included in the first line of the docstring, as it is done correctly e.g. for `nn.BatchNorm1d`.

But for `nn.ELU` it would be may be better to render an exact function in question.

### Versions

not relevant

**Comment by jbschlosser** [comment]
Hey @lithuak, thanks for the report! If you'd like to contribute, we certainly would accept PRs addressing this. As you mentioned, most likely the way we want to go here is to tweak the activation function docstrings so that the first line provides a full useful summary.

**Comment by lithuak** [comment]
Hey @jbschlosser, I can try to do that!

After taking some time looking at those docstrings I think this could work:

1) Have both the full name and the abbreviation of activation function in the first line
2) If the formula is short enough to be inlined - include it, otherwise - skip it.
3) Try to add the reference at the original paper if possible.

I'll follow up with some PR's to illustrate this approach.
